README.md

```diff
@@ -1,2 +1,101 @@
 # microcks-testcontainers-dotnet
 .NET lib for Testcontainers that enables embedding Microcks into your unit tests with lightweight, throwaway instance thanks to containers.
+
+
+## How to use it?
+
+### Include it into your project dependencies
+
+```
+dotnet add package Microcks.Testcontainers
```

Comment:
I suppose this allow to add the latest released version. How to pin a specific version? Also where the current version (in dev mode) is actually specified? I expected to see that in `Directory.Build.props` or `GitVersion.yaml` but didn't find the information...

Reply:
Yes, `dotnet add package` add the lastversion.

In your application, if you have a `.props` file, it's to customize build settings for many projects in a solution.
```
  MySolution.sln
  Directory.Build.props (1)
  \src
    Directory.Build.props (2)
    \Project1
    \Project2
  \test
    Directory.Build.props (3)
    \Project1Tests
    \Project2Tests
```
1: Apply to Projet1, Project2, Project1Tests and Project2Tests
2: Apply to Project1 and Project2, 
3: Apply to  Project1Tests and Project2Tests

It depends on the project.


For the **package**, the next version is calculate with GitVersion tool. 
Here, GitVersion determine the version with "Conventionnal Commit" (GitVersion.yml is a config file).
- feat: 0.**y + 1**.0
- feat!: **x +1**.0.0 (or breaking change in commit message).

I will edit the readme to add `dotnet add package Microcks.Testcontainers --version X.Y.Z`

Reply:
Ok got it but how current version is specified or should be specified during the release process? Typically, the "first" release would be `0.1.0` to follow same logic that other languages. How will we do that?

Reply:
It's depends if you "squash" or "rebase" my PR (with "feat: " in title).

Now, it's 0.1.5 (I had a new commit "chore").

I have 1 "feat" => minor + 1
I have: 1 docs, 1 ci, 3 chore => patch + 5.

For more information: https://www.conventionalcommits.org/en/v1.0.0/

It's possible to specify "next-version:" in gitversion.yaml and/or change workflow (github, gitflow or trunkbased).
Or simply don't used gitversion for an another tool / script. 

I use it in java, C#, etc ... but if you want to use an another solution it's ok for me.

Reply:
Ok, i can remove gitversion and add version in directory.build. It's ok for you ?

Reply:
For Release Drafter, I don't know if he can get the version number otherwise.

Reply:
> Ok, i can remove gitversion and add version in directory.build. It's ok for you ?

Let's do that for now.

Reply:
Before release, remove VersionSuffix value in Directory.Build.props.
After the release is done, increase the version number in VersionPrefix, and add "alpha" in VersionSuffix.
This ensures that the next dev release will have a higher version number than the latest release.

This two action builds a nuget package and publishes it on nuget.org. I think in the future, it's better to push the "prerelease" on github package and not on nuget.org.

In CICD, I extract value to keep to "release-drafter". I choose `VersionSuffix` & `VersionPrefix` because `Version` is formed from the combination of `VersionPrefix` and `VersionSuffix. And `Version` is the default value for all the other version (`AssemblyVersion`, `FileVersion`, etc...)

I'm adding a build number dynamically, it's best practice with nuget to not overwrite a version (you have cache in local developer machine).

I recommand to read this [documentation](https://learn.microsoft.com/en-us/dotnet/fundamentals/apicompat/package-validation/baseline-version-validator), if you increase the developer expérience in the future with "PackageValidationBaselineVersion"

Thanks

Reply:
Noted!